src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -484,4 +486,17 @@ public void close() throws IOException {
             // Else retry in case a new buf was CASed in fill()
         }
     }
+
+    @Override
+    public synchronized long transferTo(OutputStream out) throws IOException {
+        Objects.requireNonNull(out, "out");
+        int avail = count - pos;
+        if (avail > 0) {
+            out.write(buf, pos, avail);
+            count = 0;
+            pos = 0;
+        }
+        return avail + getInIfOpen().transferTo(out);
+    }
+
```

Comment:
Hi Mark, did you double check with all the existing subclasses of `BufferedInputStream` that the new implementation of `transferTo` will not cause trouble, if any of the subclasses is overriding `int read(byte[] b, int off, int len)`? If I'm not mistaken - with this new implementation of `transferTo` then `BufferedInputStream::read(byte[] b, int off, int len)` (a public method that can be overridden) will no longer be called by `transferTo`. This could be causing problems if `read` is overriden to do some special processing in the subclass.

Reply:
Thank you, good catch! I will check the subclasses contained in OpenJDK! 👍